docs/changelog/138467.yaml

```diff
@@ -0,0 +1,7 @@
+pr: 138467
+summary: "Fixed a bug that dash prefix expressions, including both concrete names\
```

Comment:
`language nit`: I'd rephrase and add examples:

_Fixed a bug where dash-prefixed expressions, for both specific index names and wildcard patterns (for example: `-system-index`, `-logs-*`), were not consistently excluded during index resolution._